### ISSUE TITLE
Add more tests for get_regressions()

### ISSUE THREAD

**Issue by klahnakoski** [mozilla/mozci#108]
In my exploration of the data while building the dashboard, I noticed a number of labels that look like regressions, but `get_regressions()` was not returning them as such.

* Make a query to point out the most egregious
* Find a way to display them
* confirm with ahal/marco-c that they *are* regressions
* make a test
* repeat

**Comment by marco-c** [comment]
Thanks, I'd love to have more regressions to feed bugbug!

**Comment by klahnakoski** [comment]
Here are some top candidates.  Let's look at one to determine if it even legitimate:

![image](https://user-images.githubusercontent.com/2334429/77685562-fde74080-6f71-11ea-8edd-bbffbdc430a3.png)

The first one starts at 109145:
*  Passes 30 previous pushes, then 
* a bad push is encountered
* test fails for next ten pushes (including bad one)
* backout of bad push is done
* after (and including) the backout, 29 pushes pass

Here is the Activedata query that shows the label's test results 

```
{
	"from":"unittest",
	"select":[
		{"value":"repo.push.date","aggregate":"max"},
		{
			"name":"pass",
			"value":{"when":"result.ok","then":1},
			"aggregate":"sum"
		},
		{
			"name":"fail",
			"value":{"when":"result.ok","else":1},
			"aggregate":"sum"
		}
	],
	"groupby":"repo.push.id",
	"where":[
		{"eq":{"run.key":"test-android-em-7.0-x86_64/opt-geckoview-mochitest-plain-e10s-4"}},
		{"gte":{"repo.push.id":109000}}
	],
	"sort":"repo.push.id",
	"limit":1000
}
```

**Comment by marco-c** [comment]
Copying conversation from Matrix.

**marco**
https://treeherder.mozilla.org/#/jobs?repo=autoland&revision=f848fd145194dea1bee4f54660ad6b6aa8c4b6f9
so this case is easy: they are missed by mozci because they are marked as fixed by commit
https://hg.mozilla.org/integration/autoland/rev/89826f85b5bd69574f7e268b67c24487d70d1d95
and this commit is a backout of another push
now, in this case the sheriffs' classification was wrong
but I would hope that usually we can rely on sheriffs' classifications
this applies to the first 4 in your table
let me check the others
the fifth in your table refers to https://treeherder.mozilla.org/#/jobs?repo=autoland&revision=9e74186c74f2263dcd5784bff180893cb395c076&searchStr=devtools-chrome&selectedJob=293307737
here the classification is correct, but it contains the bug number and the commit for some reason
instead of just the commit
so I guess we failed to "parse" the classification
the sixth in your table refers to https://treeherder.mozilla.org/#/jobs?repo=autoland&revision=fac88f2cec9c34349e1f6742a41bfd0002bf3ae0&searchStr=geckoview-mochitest&selectedJob=292736223
the sheriffs marked it as intermittent, and it looks correct

**ekyle**
the shariff marking is correct or the fact it is a regression is correct?

**marco**
actually partly both :P

**ekyle**
:)

**marco**
there is both an intermittent failure and a real failure in that task
so I guess the sheriff classification was wrong
I mean, I would have chosen the real failure over the intermittent one
the last one was also wrong classification by sheriff as intermittent

**Comment by marco-c** [comment]
I've checked the fifth push in the table to see if we could somehow handle the weird classification note (maybe it's used for other failures too, you never know). The classification on Treeherder is `fixed by commit 1424914 b76e1041bd833e1169d699edf160b7c4b7e82051`, in ActiveData it's `intermittent`. Any idea why there would be a mismatch?

**Comment by klahnakoski** [comment]
Is this a regression?

https://treeherder.mozilla.org/#/jobs?repo=autoland&tier=1%2C2%2C3&revision=1da2a939d9cf39156b38b639f77d777b0821eb4e&selectedJob=292457872

[9,4,2] - passed 9 times (zero failures), in the past 30 pushes the bad push; failed 4 times (no passes) in all pushes between the bad push and the backout; passed 2 (zero failure) in 30 pushes following the backout

**Comment by marco-c** [comment]
Yes, I think so: Wr3 and Wr5 were correctly classified, Wr1 was not correctly classified.

**Comment by klahnakoski** [comment]
is this a regression for push 05881652e9406c4ae2a2609f7c1d40c6efcf11ef?  

`test-android-hw-p2-8-0-android-aarch64-qr/debug-geckoview-reftest-e10s-1`
 was not run on the bad push, but 

[2,2,5] - passed 2 times (zero failures), in the 30 pushes before the bad push; failed 2 times (no passes) in all pushes between the bad push and the backout; passed 5 (zero failure) in 30 pushes following the backout

Here are the two failures:

* https://treeherder.mozilla.org/#/jobs?repo=autoland&revision=8d63082191f73864cd48fa4cfbdd4dfca26a0600&selectedJob=293123101
* https://treeherder.mozilla.org/#/jobs?repo=autoland&revision=9a7c2d7af71fde70591bf77a05bfb4f40e31c86b&selectedJob=293123720

Here are all the pushes from bad to backout:

* https://treeherder.mozilla.org/#/jobs?repo=autoland&revision=05881652e9406c4ae2a2609f7c1d40c6efcf11ef
* https://treeherder.mozilla.org/#/jobs?repo=autoland&revision=73fb48251c2a401447e3abc1c8fd88f6adc3016b
* https://treeherder.mozilla.org/#/jobs?repo=autoland&revision=c4dc09959bede87ae17f28283fa7eae9223fd3bf
* https://treeherder.mozilla.org/#/jobs?repo=autoland&revision=8d63082191f73864cd48fa4cfbdd4dfca26a0600
* https://treeherder.mozilla.org/#/jobs?repo=autoland&revision=9a7c2d7af71fde70591bf77a05bfb4f40e31c86b
* https://treeherder.mozilla.org/#/jobs?repo=autoland&revision=f4c76e53627976bede4bd715229484fac6512aa3
* https://treeherder.mozilla.org/#/jobs?repo=autoland&revision=30b275cb80cda8e612dc9f8c74fe8ba4a3c154cd
* https://treeherder.mozilla.org/#/jobs?repo=autoland&revision=cd8f61214a62224c8e0844e1ef5407cff05d6d28

**Comment by marco-c** [comment]
> is this a regression for push 05881652e9406c4ae2a2609f7c1d40c6efcf11ef?

This one seems like a infrastructure issue: https://bugzilla.mozilla.org/show_bug.cgi?id=1622155

**Comment by klahnakoski** [comment]
> I've checked the fifth push in ... classification on Treeherder is `fixed by commit 1424914 b76e1041bd833e1169d699edf160b7c4b7e82051`, in ActiveData it's `intermittent`. Any idea why there would be a mismatch?

I do not know where you see the "intermittent" markup. The data I used ignored all intermittent tasks, and sees 110196 as fail.

**Comment by marco-c** [comment]
>> I've checked the fifth push in ... classification on Treeherder is fixed by commit 1424914 b76e1041bd833e1169d699edf160b7c4b7e82051, in ActiveData it's intermittent. Any idea why there would be a mismatch?

> I do not know where you see the "intermittent" markup. The data I used ignored all intermittent tasks, and sees 110196 as fail.

Mmmmh, what do you mean? I just told you I see it on ActiveData :P
```
{
	"select":"failure.classification",
	"from":"treeherder",
	"where":{"eq":{"run.taskcluster.id":"GmGQk363QKC9JhQGv49TPw"}}
}
```

**Comment by marco-c** [comment]
#118 would catch 5 out of the 9 problems here (1 would still not be caught because of the mismatch with the classification in ActiveData, 3 would still not be caught as they were marked as "intermittent" by sheriffs and detecting bad "intermittent" classifications is harder).
